app/src/ui/notifications/pull-request-checks-failed.tsx

```diff
@@ -0,0 +1,301 @@
+      <Dialog
+        id="pull-request-checks-failed"
+        type="normal"
+        title={dialogTitle}
+        dismissable={false}
+        onSubmit={this.props.onSubmit}
+        onDismissed={this.props.onDismissed}
+        loading={loading}
+      >
+        <DialogContent>
+          <Row>
+            <span className="summary">
+              {failedChecks.length} {pluralChecks} failed in your pull request.
+              Do you want to switch to that Pull Request now and start fixing{' '}
+              {pluralThem}?
+            </span>
+          </Row>
+          <Row>
+            <div className="ci-check-run-dialog-container">
+              <div className="ci-check-run-header">
+                <span className="message">
+                  {this.props.commitMessage.slice(0, 72)}
+                </span>
+                <span aria-hidden="true">
+                  <Octicon symbol={OcticonSymbol.gitCommit} />
+                </span>{' '}
+                <span className="sha">{this.props.commitSha.slice(0, 9)}</span>
+                {this.renderRerunButton()}
+              </div>
+              <div className="ci-check-run-content">
+                <CICheckRunList
+                  checkRuns={this.state.checks}
+                  loadingActionLogs={this.state.loadingActionLogs}
+                  loadingActionWorkflows={this.state.loadingActionWorkflows}
+                  showLogsInline={false}
+                  selectable={true}
+                  selectedCheckRun={selectedCheck}
+                  baseHref={baseHref}
+                  onViewOnGitHub={this.onViewOnGitHub}
+                  onCheckRunClick={this.onCheckRunClick}
+                />
+                {selectedCheck !== undefined && (
+                  <CICheckRunLogs
+                    checkRun={selectedCheck}
+                    loadingActionLogs={this.state.loadingActionLogs}
+                    loadingActionWorkflows={this.state.loadingActionWorkflows}
+                    baseHref={baseHref}
+                  />
+                )}
+              </div>
+            </div>
+          </Row>
+        </DialogContent>
+        <DialogFooter>
+          <OkCancelButtonGroup
+            onCancelButtonClick={this.props.onDismissed}
+            cancelButtonText="Dismiss"
+            okButtonText={okButtonTitle}
+            onOkButtonClick={this.onSubmit}
+          />
+        </DialogFooter>
+      </Dialog>
+    )
+  }
+
+  public componentDidMount() {
+    this.loadCheckRunLogs()
+  }
+
+  public componentWillUnmount() {
+    this.checkRunsLoadCancelled = true
+  }
+
+  private renderRerunButton = () => {
+    const { checks } = this.state
+    return (
+      <div className="ci-check-rerun">
+        <Button onClick={this.rerunJobs} disabled={checks.length === 0}>
+          <Octicon symbol={syncClockwise} /> Re-run jobs
+        </Button>
+      </div>
+    )
+  }
+
+  private rerunJobs = () => {
+    // Get unique set of check suite ids
```

Comment:
Hmm.. would you mind creating a dispatcher method that we pass the checks and the githubrepository into to handle this duplicated logic? 

This method will evolve some with the actions team adding the property to determine which checks can be retried and then being able to only rerun failed jobs. It would be nice that if we update the dispatcher method it will update both implementations that I think we would expect similar behavior from. :)